src/excalidraw-backend/server.ts

```diff
@@ -311,18 +316,25 @@ export class Server {
       for await (const _ of setInterval(this.saveIntervalMs, null, {
         signal: ac.signal,
       })) {
-        const saved = await this.sendSaveMessage(roomId, {
+        const data = await this.sendSaveMessage(roomId, {
           timeout: this.saveTimeoutMs,
         });
 
-        if (saved === undefined) {
+        if (data === undefined) {
           this.logger.verbose?.(
             `No eligible sockets found to save '${roomId}'`,
           );
-        } else if (saved) {
-          this.logger.verbose?.(`Saving '${roomId}' successful`);
+        } else if (data.saved) {
+          this.logger.verbose?.(
+            `Room '${roomId}' saved successful by '${data.socket.data.userInfo.email}'`,
+          );
         } else {
-          this.logger.error(`Saving '${roomId}' failed`);
+          data.socket.data.failedSaves++;
+          data.socket.data.canSave =
+            data.socket.data.failedSaves === this.saveConsecutiveFailedAttempts;
```

Comment:
are you sure about this condition? I haven't tested it, but canSave should be `= failedSaves < saveConsecutiveFailedAttempts` no?

Reply:
yes, I have missed that